### ISSUE TITLE
In Ivy, SkipSelf() is not accessing the parent's ViewContainerRef, ElementRef, etc

### ISSUE THREAD

**Issue by Shijir** [angular/angular#34066]
# 🐞 bug report

### Affected Package

 The issue is caused by Angular Ivy

### Description

Before Ivy, I was able to access the `ViewContainerRef`, `ElementRef`, etc of the parent directive using the `@SkipSelf` decorator in the child directive. With Ivy integrated in my project, it's no longer skipping its own injections and accessing the ones in its parent. 

For example, I have the following code: 

In app.component.ts
```ts
@Directive({ selector: '[parentDirective]' })
export class ParentDirective {
  constructor(private vc: ViewContainerRef){
    console.log(vc.element);
  }
}

@Directive({ selector: '[childDirective]' })
export class ChildDirective {
  constructor(@SkipSelf() private vc: ViewContainerRef){
    console.log(vc.element);
  }
}
```
in app.component.html

```html
<div parentDirective>parent <span childDirective>child</span></div>
```

These two directive classes should log the same ViewContainerRef element. However, with Ivy, you can see that's no longer the case. Here is the [reproduction](https://ivy.ng-run.com/edit/xMfu7xoF1t4b8hFvoU9N?open=app%2Fapp.component.html). This reproduction is made in Angular Ivy 8.1.0-next.3. But I also see this behavior in my project that's on "@angular/core": "9.0.0-rc.3",

## 🔬 Minimal Reproduction

[Minimal Reproduction](https://ivy.ng-run.com/edit/xMfu7xoF1t4b8hFvoU9N?open=app%2Fapp.component.html)

## 🌍  Your Environment

**Angular Version:**
<pre><code>

"@angular/core": "9.0.0-rc.3",

</code></pre>

**Comment by kara** [comment]
Note: When trying to fix this, we realized that it's non-trivial because of the way Ivy is architected. We should consider whether this is actually a feature we want long-term due to the design / performance cost.

**Comment by johncrim** [comment]
@kara - RE "whether this is actually a feature we want long-term due to the design / performance cost" - Please see my comments on #39475

In short, [HostBindings not updated by detectChanges() in OnPush component](22560) (or detached components) is a real problem, and the workaround for that requires accessing the parent view's ChangeDetector. 

Personally I think the HostBindings update design is a root problem (it should be updateable without jumping through hoops or understanding things like "hostbindings aren't updated by a view's detectChanges()") - I don't know how many other common use-cases there are requiring core parameter types from the node's parent.

**Comment by AndrewKushnir** [comment]
FYI, this issue is resolved by #39464 and the fix will be available in v11.0.0. Thank you.
